NUSGrabberGUI/Form1.cs

```diff
@@ -191,49 +199,60 @@ private void STTitleList_SelectedIndexChanged(object sender, EventArgs e)
             {
                 if (v != null && v != "v0" && v != "" && v != " ")
                 {
-                    try { STVersionList.Items.Add(v.TrimStart(' ').Substring(0, v.LastIndexOf('(') - 1)); }
+                    try { STVersionList.Items.Add(v.TrimStart(' ').Substring(0, (v.LastIndexOf('(') > -1 ? v.LastIndexOf('(') : v.Length) - 1)); }
                     catch { STVersionList.Items.Add(v.TrimStart(' ')); }
                 }
             }
             STVersionList.SelectedIndex = 0;
             try { STTitleIDLabel.Text = "Title ID: " + (STTitleList.SelectedItem as ListItem).Title_ID.ToString(); }
             catch { }
+            STVersionList.EndUpdate();
         }
 
         private void STSearchBox_TextChanged(object sender, EventArgs e)
         {

```

Comment:
It looks like there were a lot of changes here however it's just the try catch that made the git diff to look huge, all I did was to add a try/catch block to block the UI and the beginupdate()/endupdate() calls.